src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -157,19 +166,23 @@ def error(self, message):
         recommender.set_help_examples(self.get_examples(self.prog))
         recommendations = recommender.provide_recommendations()
 
-        az_error = ArgumentUsageError(message)
+        # Get command to provide to error help
+        full_command = self.prog + ' ' + self.full_raw_command
+
+        az_error = ArgumentUsageError(message, full_command)
         if 'unrecognized arguments' in message:
-            az_error = UnrecognizedArgumentError(message)
+            az_error = UnrecognizedArgumentError(message, full_command)
         elif 'arguments are required' in message:
-            az_error = RequiredArgumentMissingError(message)
+            az_error = RequiredArgumentMissingError(message, full_command)
         elif 'invalid' in message:
-            az_error = InvalidArgumentValueError(message)
+            az_error = InvalidArgumentValueError(message, full_command)
 
         if '--query' in message:
             from azure.cli.core.util import QUERY_REFERENCE
             az_error.set_recommendation(QUERY_REFERENCE)
         elif recommendations:
             az_error.set_aladdin_recommendation(recommendations)
+        az_error.request_error_assistance(cli_ctx)
```

Comment:
Why does this method need to be called twice?  
https://github.com/Azure/azure-cli/pull/27801/files#r1390560463